template_handler/cdk.py

```
@@ -0,0 +1,199 @@
+        credentials: Credentials = self.connection_manager._get_session(
+            self.connection_manager.profile,
+            self.connection_manager.region,
+            self.connection_manager.iam_role
+        ).get_credentials()
+        envs.update(
+            AWS_ACCESS_KEY_ID=credentials.access_key,
+            AWS_SECRET_ACCESS_KEY=credentials.secret_key,
+        )
+
+        # There might not be a session token, so if there isn't one, make sure it doesn't exist in
+        # the envs being passed to the subprocess
+        if credentials.token is None:
+            envs.pop('AWS_SESSION_TOKEN', None)
+        else:
+            envs['AWS_SESSION_TOKEN'] = credentials.token
+
+        return envs
+
+    def _cmd_exists(self, cmd) -> bool:
+        """
+        Checks whether a specified CLI command exists
+        """
+
+        cmd_exists = shutil.which(cmd) is not None
+        self.logger.debug(f"command '{cmd}' exists: {cmd_exists}")
+        return cmd_exists
+
+    def _node_package_exists(self, package) -> bool:
+        """
+        Checks whether a specifie Node package exists either in the workspace or global scope
+        """
+
+        package_exists = False
+        workspace_result = subprocess.run(
+            f'npm list {package}',
+            shell=True,
+            capture_output=True
+        )
+        self.logger.debug(f"Workspace NPM package '{package}' exists: {not bool(workspace_result.returncode)}")
+
+        if workspace_result.returncode == 0:
+            package_exists = True
+        else:
+            global_result = subprocess.run(
+                f'npm --global list {package}',
+                shell=True,
+                capture_output=True
+            )
+            self.logger.debug(f"Global NPM package '{package}' exists: {not bool(global_result.returncode)}")
+
+            if global_result.returncode == 0:
+                package_exists = True
+
+        return package_exists
```

This might be a bit overkill. If you run `subprocess` with `check=True`, any nonzero exit code will result in an exception being raised. In that case, I think it would just make sense to do the thing you want to do and let it blow up with a relevant exception if it can't instead of doing all this checking beforehand.

---

You're right. It's not a big deal.